### DISCUSSION TITLE
bash to zsh

### DISCUSSION THREAD

**Discussion by AnthonyBAC** [mylinuxforwork/dotfiles#292] [General]
Hi, i leave zsh as default with the help of the welcome application, now I have a question, is it possible to have the same configuration that bash has in zsh in terms of visuals, one of the things is that bash takes the wallpaper colors as default and Another is the fasfetch that looks different, in this sense I must say that I am totally ignorant. I tried to review both ~/.bashrc and ~/.zshrc files but they seemed similar to me. Is there something I'm not doing right? thanks in advance

![image](https://github.com/user-attachments/assets/1979803e-5369-47de-9924-a90f678b4465)

**Comment by alantmwong** [comment]
Had you found any solution? I'm trying to figure this out as well

**Reply by AnthonyBAC** [reply]
Are you usging default dotfiles, ml4w or something like that?

**Reply by alantmwong** [reply]
Yes, I’m using default dotfiles and ml4w

**Comment by AnthonyBAC** [comment]
first enter with 
sudo nano or vim to ~/.zshrc 
and check that it has these lines of code, the first is for pywal to connect and work correctly and the second is for connecting to the .config files sorry for my english
----
cat ~/.cache/wal/sequences
----
for f in ~/.config/zshrc/*; do
    if [ ! -d $f ] ;then
        c=`echo $f | sed -e "s=.config/zshrc=.config/zshrc/custom="`
        [[ -f $c ]] && source $c || source $f
    fi
done`

and for the fastfetch lo a like, yo need to enter on .config/zshrc to edit the 00-init an paste this lines 

if [[ $(tty) == *"pts"* ]]; then
    fastfetch --config examples/13
else
    echo
    if [ -f /bin/qtile ]; then
        echo "Start Qtile X11 with command Qtile"
    fi
    if [ -f /bin/hyprctl ]; then
        echo "Start Hyprland with command Hyprland"
    fi
fi


I'm still learning, the only solution I had was to look for the bash configuration, and see what each thing works for, currently this is what worked for me I hope it helps you

**Reply by alantmwong** [reply]
Hi, thank you for the help. I followed your advice but I can't seem to figure out how to change the configured font colour. I want to change the font colour in zsh. I've checked in ~/dotfiles/.config/zshrc/20-customization but can't seem to find what exactly can help me change the font colour. This is the contents of my ~/dotfiles/.config/zshrc/20-customization file:

# -----------------------------------------------------
# CUSTOMIZATION
# -----------------------------------------------------
POSH=agnoster

# -----------------------------------------------------
# oh-myzsh themes: https://github.com/ohmyzsh/ohmyzsh/wiki/Themes
# -----------------------------------------------------
#ZSH_THEME=robbyrussell

# -----------------------------------------------------
# oh-myzsh plugins
# -----------------------------------------------------
plugins=(
    git
    sudo
    web-search
    archlinux
    zsh-autosuggestions
    zsh-syntax-highlighting
    fast-syntax-highlighting
    copyfile
    copybuffer
    dirhistory
)

# Set-up oh-my-zsh
source $ZSH/oh-my-zsh.sh

# -----------------------------------------------------
# Set-up FZF key bindings (CTRL R for fuzzy history finder)
# -----------------------------------------------------
source <(fzf --zsh)

# zsh history
HISTFILE=~/.zsh_history
HISTSIZE=10000
SAVEHIST=10000
setopt appendhistory

# -----------------------------------------------------
# Prompt
# -----------------------------------------------------
eval "$(oh-my-posh init zsh --config $HOME/.config/ohmyposh/zen.toml)"

# Shipped Theme
#eval "$(oh-my-posh init zsh --config /usr/share/oh-my-posh/themes/agnoster.omp.json)"

**Reply by alantmwong** [reply]
Specifically, the first screenshot is the zsh font colour, the second is the bash font colour

![image](https://github.com/user-attachments/assets/862a3894-e9a8-41c1-91f6-ded693f647d8)
![image](https://github.com/user-attachments/assets/c6b048ce-8bac-4e1f-bcbc-78a8c5855439)

**Reply by AnthonyBAC** [reply]
did you install anything else or you just change to zsh ?

**Reply by alantmwong** [reply]
I just change to zsh. I think what zsh does, is that it automatically changes font colour because of the colour palette. So, since my wallpaper background has a particular colour palette, zsh changes the colours based on that

**Reply by AnthonyBAC** [reply]
That's what I was thinking, that's with the first line of code that I gave you, which takes the colors of your wallpaper with pywal, did you try with another wallpaper?

**Reply by mylinuxforwork** [reply]
@AnthonyBAC the zsh config includes several plugins that have an effect on the terminal text color.  Autocomplete and syntax highlighting.

**Reply by alantmwong** [reply]
Thanks both. I’ve disabled the plugins you mentioned and now the font colour how I like it! :) it’s a pleasure using the ml4w dotfiles Stephan!

**Reply by mylinuxforwork** [reply]
Great.
